lib/puppet/provider/grafana_datasource/grafana.rb

```diff
@@ -63,6 +63,12 @@ def datasources
 
         datasource = JSON.parse(response.body)
 
+        basic_auth_password = if datasource.key?('secureJsonData') && datasource['secureJsonData'].key?('basicAuthPassword')
```

Comment:
`secureJsonData` is never part of the response.  There's `secureJsonFields` but this just returns booleans.  See https://github.com/grafana/grafana/issues/20274#issuecomment-1064626061 :(

Reply:
:(

Reply:
We don't use basic auth so I didn't verify that part, but that's bummer. We are using this PR internally until something can be figured out here.